### ISSUE TITLE
Utf8JsonWriter microbenchmark does not compile on corefx master

### ISSUE THREAD

**Issue by watfordgnf** [dotnet/performance#560]
Using ff6a7d840391c0fb41ec72a41a41aa9e32841eec against [corefx master](https://github.com/dotnet/corefx/commit/28efaabc4349d11a95f8f31eb932c84f207916ab):
```
C:\source\performance\src\benchmarks\micro>dotnet run -c Release -f netcoreapp3.0 --coreRun "C:\source\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" --filter System.Collections* --artifacts "c:\scratch\corefxperf\before"
corefx\System.Text.Json\Utf8JsonWriter\Perf.Base64.cs(15,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.Basic.cs(24,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.Booleans.cs(16,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.Ctor.cs(14,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.DateTimes.cs(16,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.Deep.cs(19,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.Doubles.cs(16,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.Guids.cs(16,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Utf8JsonWriter\Perf.Strings.cs(16,17): error CS0246: The type or namespace name 'ArrayBufferWriter<>' could not be found (are you missing a using directive or an assembly reference?) [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]

The build failed. Please fix the build errors and run again.
```

It looks like perhaps a local corefx build needs to define `MAKE_ABW_PUBLIC`?

https://github.com/dotnet/corefx/blob/3ea7b55e275fb6367f67e7fb0432be4bb3902f09/src/Common/src/System/Buffers/ArrayBufferWriter.cs#L12-L18

**Comment by adamsitnik** [comment]
Could you please share the output of `dotnet --version`? Most probably you don't have the latest .NET Core 3.0 preview installed.

We have a nice python3 script that can download it for you and run the benchmarks:

```
git clone https://github.com/dotnet/performance
cd performance
py .\scripts\benchmarks_ci.py -f netcoreapp3.0 --corerun "C:\source\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" --filter System.Collections* --bdn-artifacts "c:\scratch\corefxperf\before"
```

**Comment by watfordgnf** [comment]
I was pending a restart after both VS 16.2.0-pre.2.0 and 3.0.100-preview6-012264. I receive the following (without running the python script):
```
corefx\System.Text.Json\Serializer\WriteJson.cs(29,60): error CS1503: Argument 2: cannot convert from 'T' to 'System.IO.Stream' [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Serializer\WriteJson.cs(45,60): error CS1503: Argument 2: cannot convert from 'T' to 'System.IO.Stream' [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
corefx\System.Text.Json\Serializer\ReadJson.cs(34,60): error CS1503: Argument 2: cannot convert from 'T' to 'System.IO.Stream' [C:\source\performance\src\benchmarks\micro\MicroBenchmarks.csproj]

The build failed. Fix the build errors and run again.
```
Running your script however, shows me that the latest preview is 7. Does this mean that even if I'm using local builds of corefx I should be using the `benchmarks_ci.py` script? (I presume that this is how I get the latest tools/SDK for *compilation* of the benchmarks)

**Comment by adamsitnik** [comment]
In this repository, we are using the latest builds of .NET Core 3.0 which you can download from https://github.com/dotnet/core-setup/.

> Running your script however, shows me that the latest preview is 7. Does this mean that even if I'm using local builds of corefx I should be using the benchmarks_ci.py script?

Yes, the easiest way to get it working is to use the python script.

**Comment by watfordgnf** [comment]
I appreciate it, perhaps the bit about the `benchmarks_ci.py` script could be moved up higher in the workflow wiki? Or perhaps a note about core-setup in the Prerequisites.

**Comment by adamsitnik** [comment]
> a note about core-setup in the Prerequisites

Good point, I have added it in #622

**Comment by watfordgnf** [comment]
Will be closed by #622.
